src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -601,6 +616,24 @@ void IJoinableTaskDependent.OnDependencyRemoved(IJoinableTaskDependent joinChild
         {
         }
 
+        /// <summary>
+        /// Gets the asynchronous task that completes when the async operation completes, if one has been created.
+        /// </summary>
+        /// <param name="task">The value of <see cref="Task"/>, or <see langword="null"/> if the task is not yet created.</param>
+        /// <returns><see langword="true"/> if the task has been created; otherwise, <see langword="false"/>.</returns>
+        internal bool TryGetTask([NotNullWhen(true)] out Task? task)
+        {
+            var wrappedTask = this.wrappedTask;
```

Comment:
In practice it doesn't make a difference. Both paths produce a correct value, but one produces a correct value without locks or allocations.

Reply:
This sounds like a reply to a different comment, no?

Reply:
It was intended here. The two paths are:

* A value was read that is the same as `Volatile.Read` would produce
* A value was read that is different than `Volatile.Read` would produce

Reply:
In reviewing this, I've deleted this method because we don't need it, evidently. The `IsCompleted` property can simply depend on `IsCompleteRequested` (which by the way uses volatile and no locks).